src/MongoDB.Bson/BsonDefaults.cs

```diff
@@ -58,30 +76,64 @@ public static IBsonSerializer DynamicArraySerializer
         /// </summary>
         public static IBsonSerializer DynamicDocumentSerializer
         {
-            get 
+            get
             {
                 if (!__dynamicDocumentSerializerWasSet)
                 {
                     __dynamicDocumentSerializer = BsonSerializer.LookupSerializer<ExpandoObject>();
                 }
-                return __dynamicDocumentSerializer; 
+                return __dynamicDocumentSerializer;
             }
-            set 
+            set
             {
                 __dynamicDocumentSerializerWasSet = true;
-                __dynamicDocumentSerializer = value; 
+                __dynamicDocumentSerializer = value;
             }
         }
 
         /// <summary>
-        /// Gets or sets the default representation to be used in serialization of 
-        /// Guids to the database. 
-        /// <seealso cref="MongoDB.Bson.GuidRepresentation"/> 
+        /// Gets or sets the default representation to be used in serialization of
+        /// Guids to the database.
+        /// <seealso cref="MongoDB.Bson.GuidRepresentation"/>
         /// </summary>
+        [Obsolete("Configure serializers instead.")]
         public static GuidRepresentation GuidRepresentation
         {
-            get { return __guidRepresentation; }
-            set { __guidRepresentation = value; }
+            get
+            {
+                if (BsonDefaults.GuidRepresentationMode != GuidRepresentationMode.V2)
+                {
+                    throw new InvalidOperationException("BsonDefaults.GuidRepresentation can only be used when BsonDefaults.GuidRepresentationMode is V2.");
+                }
+                return __guidRepresentation;
+            }
+            set
+            {
+                if (BsonDefaults.GuidRepresentationMode != GuidRepresentationMode.V2)
+                {
+                    throw new InvalidOperationException("BsonDefaults.GuidRepresentation can only be used when BsonDefaults.GuidRepresentationMode is V2.");
+                }
+                __guidRepresentation = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the default representation to be used in serialization of
+        /// Guids to the database.
+        /// <seealso cref="MongoDB.Bson.GuidRepresentation"/>
+        /// </summary>
+        [Obsolete("This property will be removed in a later release.")]
+        public static GuidRepresentationMode GuidRepresentationMode
+        {
+            get { return __guidRepresentationMode; }
+            set
+            {
+                __guidRepresentationMode = value;
+                if (value == GuidRepresentationMode.V3)
+                {
+                    __guidRepresentation = GuidRepresentation.Unspecified;
```

Comment:
When mode is V3 representation MUST be Unspecified... so setting that here ensures that.

Agreed that it's a bit unusual for setting one property to result in another changing also, but it would be more dangerous not to do this.

Reply:
It's not the best design, but since it's almost temporal solution, I'm ok with it